vaadin-grid-pro-flow/src/main/java/com/vaadin/flow/component/gridpro/GridPro.java

```diff
@@ -102,10 +106,23 @@ private void setup() {
         addItemPropertyChangedListener(e -> {
             EditColumn<E> column = (EditColumn<E>) this.idToColumnMap.get(e.getPath());
 
-            column.getItemUpdater().accept(e.getItem(), e.getSourceItem().get(e.getPath()).asString());
+            if (column.getEditorType().equals("custom")) {
+                column.getItemUpdater().accept(e.getItem(), null);
+            } else {
+                column.getItemUpdater().accept(e.getItem(), e.getSourceItem().get(e.getPath()).asString());
+            }
+
 
             getDataProvider().refreshItem(e.getItem());
         });
+
+        addCellEditStartedListener(e -> {
+            EditColumn<E> column = (EditColumn<E>) this.idToColumnMap.get(e.getPath());
+
+            if (column.getEditorType().equals("custom")) {
+                ((HasValue) column.getEditorField()).setValue(column.getValueProvider().apply(e.getItem()));
```

Comment:
Avoid casting: make the type of `editorField` be `AbstractField`

Reply:
Changed.